requirements.py

```diff
@@ -0,0 +1,197 @@
+import argparse
+import json
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser(
+    description="Install OpenSight dependencies based on required feature-sets"
+)
+parser.add_argument(
+    "-y",
+    "--yes",
+    dest="autoconfirm",
+    action="store_true",
+    help="automatically confirm all prompts",
+)
+parser.add_argument(
+    "-l",
+    "--list",
+    dest="list",
+    action="store_true",
+    help="list packages to be installed and exit",
+)
+parser.add_argument(
+    "-r",
+    "--requirements",
+    dest="requirements",
+    default="requirements.txt",
+    help="specify alternative requirements.txt",
+)
+parser.add_argument(
+    "-f",
+    "--filename",
+    dest="requirements_extra",
+    default="requirements_extra.json",
+    help="specify alternative requirements_extra.txt",
+    type=argparse.FileType(),
+)
+parser.add_argument(
+    "-n",
+    "--no-requirements",
+    dest="exclude_requirements",
+    action="store_true",
+    help="exclude requirements.txt",
+)
+parser.add_argument(
+    "-e", "--exclude", dest="excludes", action="append", help="exclude features",
+)
+
+
+def get_default_overlays():
+    defaults = ["base"]
+    if sys.platform.startswith("linux"):
+        defaults.append("linux")
+    return defaults
+
+
+parser.add_argument(
+    "overlays",
+    action="append",
+    nargs="*",
+    help="overlays to install",
+    default=get_default_overlays(),
+)
```

Comment:
This should be moved above `get_default_overlays`.

Reply:
That won't work because the function is called here. `parser.add_argument` default parameter is not a function that returns a value, I just call a function to have a local scope where I create the default value.